apps/sms/test/unit/message_manager_test.js

```diff
@@ -461,23 +461,34 @@ suite('message_manager.js >', function() {
     setup(function() {
       this.sinon.spy(MockNavigatormozMobileMessage, 'markMessageRead');
       messageIds = [1, 2, 3];
-
-      MessageManager.markMessagesRead(messageIds);
     });
 
     test('properly mark all ids as read', function() {
+      MessageManager.markMessagesRead(messageIds);
       while (MockNavigatormozMobileMessage.mTriggerMarkReadSuccess()) {
       }
 
       sinon.assert.callCount(MockNavigatormozMobileMessage.markMessageRead, 3);
       messageIds.forEach(function(id) {
         sinon.assert.calledWith(
-          MockNavigatormozMobileMessage.markMessageRead, id
+          MockNavigatormozMobileMessage.markMessageRead, id, true
         );
       });
     });
 
+    test('properly mark all ids as unread', function() {
+      MessageManager.markMessagesRead(messageIds,/*isRead*/ false);      
+      while (MockNavigatormozMobileMessage.mTriggerMarkReadSuccess()) {
+      }
+
+      sinon.assert.callCount(MockNavigatormozMobileMessage.markMessageRead, 1);
+      sinon.assert.calledWith(
+        MockNavigatormozMobileMessage.markMessageRead, 3, false
```

Comment:
nit: This test looks fragile becasue it might be failed if we change the message id array. In this test you should verify markMessageRead is called with last element in id array, not "3"